src/EigenvalueSolverTypes.f90

```diff
@@ -586,7 +630,159 @@ SUBROUTINE init_EigenvalueSolverType_Anasazi(solver,MPIEnv,Params)
+      CALL validParams%get('EigenvalueSolverType->tolerance',tol)
+      CALL validParams%get('EigenvalueSolverType->max_iterations',maxit)
+
+      IF(.NOT. solver%isInit) THEN
+        IF(n < 1) THEN
+          CALL eEigenvalueSolverType%raiseError('Incorrect input to '// &
+            modName//'::'//myName//' - Number of values (n) must be '// &
+              'greater than 0!')
+        ELSE
+          solver%n=n
+        ENDIF
+
+        IF((nlocal < 1) .AND. (nlocal > n)) THEN
+          CALL eEigenvalueSolverType%raiseError('Incorrect input to '// &
+            modName//'::'//myName//' - Number of values (nlocal) must be '// &
+              'greater than 0 and less than or equal to (n)!')
+        ENDIF
+
+        IF(tol<=0.0_SRK) THEN
+          CALL eEigenvalueSolverType%raiseError('Incorrect input to '// &
+            modName//'::'//myName//' - Tolerance must be '// &
+              'greater than 0!')
+        ELSE
+          solver%tol=tol
+        ENDIF
 
+        IF(maxit < 1) THEN
+          CALL eEigenvalueSolverType%raiseError('Incorrect input to '// &
+            modName//'::'//myName//' - Maximum Iterations must be '// &
+              'greater than 0!')
+        ELSE
+          solver%maxit=maxit
+        ENDIF
+
+        solver%TplAnasazi = ParameterList("anasazi")
+
+ 
+        IF(Params%has('EigenvalueSolverType->anasazi_options')) THEN
+          CALL Params%get('EigenvalueSolverType->anasazi_options', anasaziParams)
+          CALL anasaziParams%toForTeuchosPlist(solver%TplAnasazi)
+          !CALL Anasazi_Init_Params(solver%eig, plID)
+        ELSE
+          !CALL Params%toForTeuchosPlist(solver%TplAnasazi)
+          CALL solver%TplAnasazi%set('Solver Type','Generalized Davidson')
+          sublist = solver%TplAnasazi%sublist('Generalized Davidson')
+          
+          CALL sublist%set('Which','LM')
+          CALL sublist%set('Convergence Tolerance',1.0d-7)
+          CALL sublist%set('Maximum Subspace Dimension',25)
+          CALL sublist%set('Restart Dimension',5)
+          CALL sublist%set('Maximum Restarts', 20)
+          
+          CALL solver%TplAnasazi%set('Preconditioner Type','MueLu')
+          sublist = solver%TplAnasazi%sublist('MueLu')
+          
+          CALL sublist%set('verbosity','none')
+          CALL sublist%set('smoother: type','RILUK')
+          CALL sublist%set('coarse: type','RILUK')
+          
+          CALL sublist%release()
+          
+        ENDIF
+        IF(solvertype/=GD) THEN
+          CALL eEigenvalueSolverType%raiseError('Incorrect input to '// &
+              modName//'::'//myName//' - Only Generalized Davidson works with Anasazi.')
+        ENDIF
+        ! create the solver handle
+        solver%eig = TrilinosEigenSolver()
+        CALL solver%eig%init(solver%Tcomm)
+        ! call solver%TplAnasazi%print()
+        ! setup the solver
+        !CALL solver%eig%setup_solver(plAnasazi)
+        !Need to set PC type
+!FIXME will punt on this for now        
```

Comment:
What is this section?  It looks like you are setting the PC earlier.. is this really needed?